pkgs/development/tools/zigmod/buildZigmodPackage.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, stdenv
+, cacert
+, git
+, zig
+, zigmod
+}:
+
+{ name ? "${attrs.pname}-${attrs.version}"
+, src
+, fetchOutputHash
+, nativeBuildInputs ? [ ]
+, postUnpack ? ""
+, preBuild ? ""
+, meta ? { }
+, ...
+} @ attrs:
+let
+  depsDir = ".zigmod";
+
+  deps = stdenv.mkDerivation {
+    inherit src;
+
+    name = "${name}-deps";
+
+    nativeBuildInputs = [
+      cacert
+      git
+      zigmod
+    ];
+
+    buildPhase = ''
+      zigmod fetch
+    '';
+
+    installPhase = ''
+      find ${depsDir} -name .git -type d -prune -exec rm -rf {} \;;
+      # Build a reproducible tar, per instructions at https://reproducible-builds.org/docs/archives/
+      tar --owner=0 --group=0 --numeric-owner --format=gnu \
+          --sort=name --mtime="@$SOURCE_DATE_EPOCH" \
+          -czf $out -C ${depsDir} .
+    '';
+
+    outputHash = fetchOutputHash;
+    outputHashMode = "recursive";
+    outputHashAlgo = "sha256";
```

Comment:
We can't hardcore this. Either we only support Sri hashes or only set this when sha256 is given.